### PR TITLE
196 new credential from builder modal

### LINKED ISSUE #196: Create a new credential from within the workflow builder
### User story
As a user that is adding a new workflow step with a credential that I haven't created yet,
I would like to be able to create a new credential from within the step,
So that I don't have to click in and out of the builder 

### Details
In the credential dropdown within the create/edit job form that is rendered on the builder, add a 'New credential' button.
When a user clicks it, they should get redirected to the 'create credential form', which on save sets this as the job credential and redirects them to the builder.

### Implementation notes

Thinking of 2 ways:

-> **Direct to credential live**

1. Create a link , live_redirect, live_patch under the credentials selector
2. Passing caller_context (current job, current workflow, current project, return_to) via params (state in url)
3. Display caller_context info in new credential live page (so user is aware of the project/workflow/job he is creating a credential for)
4. Hide project_credentials management mini module (only current project is concerned)
5. Redirect to the caller_context.return_to url with the newly created credential (return_to )
6. Consider adding a breadcrumb navigation component

-> **Open a modal with credential form component**
repeat above steps

Challenge: state flowing between pages without hurting performance (live_patch vs live_redirect)

### Release notes 
Create a new credential from within the job editor.

### Tests 
Open up workflow step, there should be a 'New credential' button below the credentials section.
When I click it the create_credential popup appears and the correct project scope is set. 
When I fill out the credential form and hit save, I am redirected to workflow page I was on before and the newly created credential is set for that step.

### QA
- [ ] This has been QA’d by Amber, branch can be merged after review

### PR DESCRIPTION
**Pull request by wanecode**
## Related issue

Fixes #196 Create a new credential from within the workflow builder

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] If needed, I've updated the changelog
- [x] Amber has QA'd this feature

### CODE REVIEW

**Comment by wanecode** [comment]
Hey @amberrignell , could you please check this feature.
- create a new credential from an existing job in the builder
- create a new credential from a new job in the builder
- create a new credential from the job form (new job button or edit a job from the job table)

**Review by taylordowns2000 — CHANGES_REQUESTED** [review]
@wanecode , please make sure that _YOU_ check your PR before requesting a review. `generated_project.yaml` does not belong in here. I think we solved this problem on `main`, you may have added a file to your commit by accident.

image

**Comment by wanecode** [comment]
> @wanecode , please make sure that _YOU_ check your PR before requesting a review. `generated_project.yaml` does not belong in here. I think we solved this problem on `main`, you may have added a file to your commit by accident.
> 
> image

**Comment by wanecode** [comment]
> > @wanecode , please make sure that _YOU_ check your PR before requesting a review. `generated_project.yaml` does not belong in here. I think we solved this problem on `main`, you may have added a file to your commit by accident.
> > image

Yes. It was an accident. Sorry

File: lib/lightning_web/live/job_live/inspector_form_component.ex [hunk 1]
```diff
@@ -145,6 +145,15 @@ defmodule LightningWeb.JobLive.InspectorFormComponent do
 
           <div class="md:col-span-1">
             <Components.Jobs.credential_select form={f} credentials={@credentials} />
+            <button
+              id="new-credential-launcher"
+              type="button"
+              phx-click={
+                Phoenix.LiveView.JS.push("new-credential", value: @job_params)
```
Comment (stuartc) [inline lib/lightning_web/live/job_live/inspector_form_component.ex:152]: I believe `phx-click="new-credential"` and `phx-value={@job_params}` should have the same effect.

I don' think we need the JS.

**Review by stuartc — CHANGES_REQUESTED** [review]
@wanecode good job overall dude! I've made a few comments that could improve things (mostly around decoupling the form from other modules).

**Pull request merged**